### PR TITLE
feat(dot-browser-selector): add Upload New File button to File Picker in New Edit Content

### LINKED ISSUE #33103: File Picker in New Edit Content Lacks Upload New File Button
### User Story

The file picker in the old edit content was recently redesigned to improve its usability, the new edit content file picker should follow its pattern but most importantly should also support the same functionality as it did before. At the moment, the File Picker in New Edit Content Lacks the Upload New File Button thus blocking users from being able to upload a new file. See video below:
https://dotcms.slack.com/files/U07LKSBNM4K/F09C3DHK90D/screen_recording_2025-08-26_at_12.44.10___pm.mov

### Acceptance Criteria

Same functionality as the old edit content in the new edit content. 

### Proposed Objective

User Experience

### Proposed Priority

Priority 1 - Show Stopper

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

#feat-edit-content

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by adrianjm-dotCMS**
## Summary

Closes #33103

Adds an **Upload New File** button to the File Picker (`dot-browser-selector`) used in New Edit Content. Users can now upload files directly to the selected site/folder without leaving the content editor.

- **Upload button** is visible in both Image Picker (`accept=image/*`) and File Picker (`accept=*/*`)
- **Button disabled** only when no site/folder is selected (`hostFolderId === ''`); System Host and real sites enable it
- **Permission handling**: button is always enabled — clicking without upload permissions shows an **inline error message** instead of disabling the button
- **Upload uses the same endpoint/service as Content Drive** (`DotUploadFileService.uploadDotAsset`)
- After a successful upload, the file list refreshes and the new file is visible but **not auto-selected**
- Upload failures (403 permissions, generic server errors) show contextual inline error messages without clearing the existing file list

https://github.com/user-attachments/assets/467d30ea-218c-40fd-b482-8066dd584dc0

## Changes

- `dot-dataview.component.ts/html` — added `$uploadDisabled` input and `[disabled]` binding on upload button
- `dot-browser-selector.component.ts/html` — added `$uploadDisabled` computed signal (`hostFolderId === ''`) and passes it to `dot-dataview`
- `dot-browser-selector.component.spec.ts` — **new spec file** covering `$uploadDisabled`, `$acceptAttr`, `onNodeSelect`, and `onFileUpload` (13 tests)

## Out of scope

- Feature 2 — Global Search across all sites/folders will be handled in a separate ticket
- - 

## Test plan

- [ ] Upload button visible in Image Picker and File Picker dialogs
- [ ] Button enabled when System Host or any site/folder is selected
- [ ] Button disabled only in the edge case where `hostFolderId` is empty
- [ ] Clicking Upload without permissions shows inline error; file list remains intact
- [ ] Happy path: select a site/folder → upload image/file → file appears in list, not auto-selected
- [ ] Generic server error shows generic inline error message
- [ ] Image Picker restricts file input to `image/*`; File Picker accepts all files

This PR fixes: #33103

### CODE REVIEW

File: core-web/libs/ui/src/lib/components/dot-browser-selector/components/dot-dataview/dot-dataview.component.ts [hunk 1]
```diff
@@ -79,4 +81,31 @@ export class DotDataViewComponent {
...
+
+    onFileSelected(event: Event): void {
+        const input = event.target as HTMLInputElement;
+        const file = input.files?.[0];
+
+        if (file) {
+            this.onUploadFile.emit(file);
+        }
+
+        input.value = '';
```
Comment (zJaaal) [inline core-web/libs/ui/src/lib/components/dot-browser-selector/components/dot-dataview/dot-dataview.component.ts]: beware of potential check detection issues

File: dotCMS/src/main/webapp/WEB-INF/messages/Language_es.properties
Comment (zJaaal) [inline dotCMS/src/main/webapp/WEB-INF/messages/Language_es.properties:1]: Why are we adding spanish keys? 
I think the translation is not handled by R&D

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/webapp/WEB-INF/messages/Language_es.properties
Comment (adrianjm-dotCMS) [reply dotCMS/src/main/webapp/WEB-INF/messages/Language_es.properties:1]: I wanted to test something with the language, and then i said like "Why not?". But you're right, if they aren't being used, I'll go ahead and remove them.

File: core-web/libs/ui/src/lib/components/dot-browser-selector/components/dot-dataview/dot-dataview.component.ts (hunk 1 above)
Comment (adrianjm-dotCMS) [reply core-web/libs/ui/src/lib/components/dot-browser-selector/components/dot-dataview/dot-dataview.component.ts]: Fixed, replaced `event.target` mutation with a `viewChild` reference ($fileInput).

**Review by zJaaal — APPROVED** [review]

**Review by dario-daza — APPROVED** [review]
